libevm/JitVM.cpp

```diff
@@ -248,47 +242,112 @@ int64_t evm_call(
 	return cost;
 }
 
+
+/// RAII wrapper for an evm instance.
+class EVM
+{
+	evm_instance* m_instance = nullptr;
+
+public:
+	EVM(evm_query_fn _queryFn, evm_update_fn _updateFn, evm_call_fn _callFn)
+	{
+		m_instance = evm_create(_queryFn, _updateFn, _callFn);
+	}
+
+	~EVM()
+	{
+		evm_destroy(m_instance);
+	}
+
+	EVM(EVM const&) = delete;
+	EVM& operator=(EVM) = delete;
+
+	struct Result : evm_result
+	{
+		Result(evm_result const& _other)
+		{
+			// FIXME: Looks like iheritance from C struct is not very effective.
```

Comment:
`Result` only provides a RAII-wrapper around `evm_result`, right? In that case, I would not use inheritance.